### PR TITLE
OCPBUGS-2657: CGU Insufficient disk space to create backup

### PR DESCRIPTION
**Pull request by sabbir-47**
- SafetyNet unit declared
- Unit test added

Signed-off-by: Sabbir Hasan <sahasan@redhat.com>

/cc @imiller0 @donpenney @jc-rh

### CODE REVIEW

File: recovery/cmd/backup_size_test.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+			freeDisk:  50.00 * 1024 * 1024 * 1024,
+			expected:  false,
+			name:      "lower disk space in same unit",
+		},
+		{
+			estimated: 80.00 * 1024 * 1024 * 1024,
+			freeDisk:  100.00 * 1024 * 1024,
+			expected:  false,
+			name:      "lower disk space in different unit",
+		},
```
Comment (donpenney) [inline recovery/cmd/backup_size_test.go:46]: Probably be good to add a case here where difference is units is at least two levels. It looks like these are all either same units or off by one (ie. G vs M), but the case that came up was terabytes of free space vs megabytes of required space. I think the original code was ok with the off-by-one (aside from the safety net issue), but had an issue with the off-by-two

File: recovery/cmd/backup_size_test.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+func TestCompare(t *testing.T) {
+
+	testcases := []struct {
+		estimated, freeDisk float64
+		expected            bool
+		name                string
+	}{
+		{
+			estimated: 80.00 * 1024 * 1024 * 1024,
```
Comment (donpenney) [inline recovery/cmd/backup_size_test.go:18]: nit: Can you add comments to each of these to explicitly call out the units/sizes for each test?

File: recovery/cmd/backup_size_test.go (hunk 1 above)
Comment (sabbir-47) [reply recovery/cmd/backup_size_test.go:64]: Added by this commit 0a94f2debff8eb3ed881ec1eab12431501f50218 as suggested

File: recovery/cmd/backup_size_test.go (hunk 2 above)
Comment (sabbir-47) [reply recovery/cmd/backup_size_test.go]: Added by this commit https://github.com/openshift-kni/cluster-group-upgrades-operator/commit/0a94f2debff8eb3ed881ec1eab12431501f50218 as suggested

**Pull request merged**